src/contrib/persistence/Akka.Persistence.Sqlite.Tests/CustomObjectSerializerSpec.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="CustomObjectSerializerSpec.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Serialization;
+using FluentAssertions;
+using Microsoft.Data.Sqlite;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Sqlite.Tests
+{
+    public class CustomObjectSerializerSpec : Akka.TestKit.Xunit2.TestKit, IAsyncLifetime
+    {
+        private static readonly Config Config = ConfigurationFactory.ParseString($@"
+akka.actor {{
+    serializers {{
+        mySerializer = ""{typeof(MySerializer).AssemblyQualifiedName}""
+    }}
+    serialization-bindings {{
+        ""System.Object"" = mySerializer
+    }}
+}}
+
+akka.persistence {{
+    journal {{
+        plugin = ""akka.persistence.journal.sqlite""
+        sqlite {{
+            connection-string = ""DataSource=AkkaJournal.db""
+            auto-initialize = on
+        }}
+    }}
+    snapshot-store {{
+        plugin = ""akka.persistence.snapshot-store.sqlite""
+        sqlite {{
+            connection-string = ""DataSource=AkkaSnapshot.db""
+            auto-initialize = on
+        }}
+    }}
+}}").WithFallback(SqlitePersistence.DefaultConfiguration());
+
+        public CustomObjectSerializerSpec(ITestOutputHelper helper) 
+            : base(Config, nameof(CustomObjectSerializerSpec), helper)
+        {
+        }
+
+        [Fact(DisplayName = "Persistence.Sql should use custom serializer for object type")]
+        public async Task CustomSerializerTest()
+        {
+            var probe = CreateTestProbe();
+            
+            var actor = Sys.ActorOf(Props.Create(() => new PersistedActor("a")));
+            actor.Tell("a", probe);
+            probe.ExpectMsg("a");
+
+            var conn = new SqliteConnection("DataSource=AkkaJournal.db");
+            conn.Open();
+            const string sql = "SELECT ej.serializer_id FROM event_journal ej WHERE ej.persistence_id = 'a'";
+            await using var cmd = new SqliteCommand(sql, conn);
+            var record = await cmd.ExecuteReaderAsync();
+            await record.ReadAsync();
+            record[0].Should().Be(9999);
```

Comment:
This fails because the payload is being serialized using the JSON serializer (1) and not MySerializer (9999)

Reply:
Could you add a comment explaining what this test actually tests - like, what use case is it here to assert? Just need a snippet at the top explaining this.